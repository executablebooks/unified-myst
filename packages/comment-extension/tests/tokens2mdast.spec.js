import { fromMarkdown } from 'mdast-util-from-markdown'

import { mystCommentMmarkExt, mystCommentMdastExt } from '../src/index.js'

const fixtures = ['%', '% ', '% abc', '% abc\n% def']

describe('Convert source to MDAST', () => {
    test.each(fixtures)(`%j`, (content) => {
        const options = {
            extensions: [mystCommentMmarkExt],
            mdastExtensions: [mystCommentMdastExt],
        }
        const mdast = fromMarkdown(content, options)
        expect(mdast).toMatchSnapshot()
    })
})
