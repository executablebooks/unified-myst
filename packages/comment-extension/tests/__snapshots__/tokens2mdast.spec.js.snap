// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Convert source to MDAST "% " 1`] = `
Object {
  "children": Array [
    Object {
      "position": Object {
        "end": Object {
          "column": 3,
          "line": 1,
          "offset": 2,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "comment",
      "value": " ",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 3,
      "line": 1,
      "offset": 2,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`Convert source to MDAST "% abc" 1`] = `
Object {
  "children": Array [
    Object {
      "position": Object {
        "end": Object {
          "column": 6,
          "line": 1,
          "offset": 5,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "comment",
      "value": " abc",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 6,
      "line": 1,
      "offset": 5,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`Convert source to MDAST "% abc\\n% def" 1`] = `
Object {
  "children": Array [
    Object {
      "position": Object {
        "end": Object {
          "column": 6,
          "line": 1,
          "offset": 5,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "comment",
      "value": " abc",
    },
    Object {
      "position": Object {
        "end": Object {
          "column": 6,
          "line": 2,
          "offset": 11,
        },
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 6,
        },
      },
      "type": "comment",
      "value": " def",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 6,
      "line": 2,
      "offset": 11,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`Convert source to MDAST "%" 1`] = `
Object {
  "children": Array [
    Object {
      "position": Object {
        "end": Object {
          "column": 2,
          "line": 1,
          "offset": 1,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "comment",
      "value": "",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 2,
      "line": 1,
      "offset": 1,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
